### ISSUE TITLE
Missing info from `argumentsError` when `code: "invalid_arguments"`

### ISSUE THREAD

**Issue by Jackman3005** [andrewvo89/zod-error#4]
Hey, nice library you have here :)

I was playing around and noticed that when I received an `invalid_arguments` code that some critical information was missing in the printout.

Here's the Zod error in JSON format:
```json
{
  "issues": [
    {
      "code": "invalid_arguments",
      "argumentsError": {
        "issues": [
          {
            "code": "invalid_type",
            "expected": "object",
            "received": "array",
            "path": [
              0
            ],
            "message": "Expected object, received array"
          }
        ],
        "name": "ZodError"
      },
      "path": [
        0,
        "renderConfig"
      ],
      "message": "Invalid function arguments"
    }
  ],
  "name": "ZodError"
}
```

I am not 100% sure, but in this case it feels like I want to recursively print the issues inside of `argumentsError` as part of the `invalid_arguments` message.

I was hoping to be able to do this on my own, but I found out that the transform function does not yield access to the issue it is processing directly.

Let me know if you have any ideas and keep up the great work!

**Comment by Jackman3005** [comment]
I guess as I think about this a bit more, I think that it would be ideal if the message catered to the exact ZodIssue type. In the case above, it's when it's a `ZodInvalidArgumentsIssue`. But there are quite a few issues (16 by my count) and I would imagine we would want to have access to the other fields in those issues as well, and likely print some/most/all(?) of them by default.

### The different Zod Issue Types
```typescript
export declare const ZodIssueCode: {
    invalid_type: "invalid_type";
    invalid_literal: "invalid_literal";
    custom: "custom";
    invalid_union: "invalid_union";
    invalid_union_discriminator: "invalid_union_discriminator";
    invalid_enum_value: "invalid_enum_value";
    unrecognized_keys: "unrecognized_keys";
    invalid_arguments: "invalid_arguments";
    invalid_return_type: "invalid_return_type";
    invalid_date: "invalid_date";
    invalid_string: "invalid_string";
    too_small: "too_small";
    too_big: "too_big";
    invalid_intersection_types: "invalid_intersection_types";
    not_multiple_of: "not_multiple_of";
    not_finite: "not_finite";
};
```

### The structure of all the different Zod Issue types (for reference)
<details>
  <summary>Structure of the different Zod issue types</summary>

  ```typescript
  export declare type ZodIssueCode = keyof typeof ZodIssueCode;
  export declare type ZodIssueBase = {
      path: (string | number)[];
      message?: string;
  };
  export interface ZodInvalidTypeIssue extends ZodIssueBase {
      code: typeof ZodIssueCode.invalid_type;
      expected: ZodParsedType;
      received: ZodParsedType;
  }
  export interface ZodInvalidLiteralIssue extends ZodIssueBase {
      code: typeof ZodIssueCode.invalid_literal;
      expected: unknown;
      received: unknown;
  }
  export interface ZodUnrecognizedKeysIssue extends ZodIssueBase {
      code: typeof ZodIssueCode.unrecognized_keys;
      keys: string[];
  }
  export interface ZodInvalidUnionIssue extends ZodIssueBase {
      code: typeof ZodIssueCode.invalid_union;
      unionErrors: ZodError[];
  }
  export interface ZodInvalidUnionDiscriminatorIssue extends ZodIssueBase {
      code: typeof ZodIssueCode.invalid_union_discriminator;
      options: Primitive[];
  }
  export interface ZodInvalidEnumValueIssue extends ZodIssueBase {
      received: string | number;
      code: typeof ZodIssueCode.invalid_enum_value;
      options: (string | number)[];
  }
  export interface ZodInvalidArgumentsIssue extends ZodIssueBase {
      code: typeof ZodIssueCode.invalid_arguments;
      argumentsError: ZodError;
  }
  export interface ZodInvalidReturnTypeIssue extends ZodIssueBase {
      code: typeof ZodIssueCode.invalid_return_type;
      returnTypeError: ZodError;
  }
  export interface ZodInvalidDateIssue extends ZodIssueBase {
      code: typeof ZodIssueCode.invalid_date;
  }
  export declare type StringValidation = "email" | "url" | "uuid" | "regex" | "cuid" | "cuid2" | "datetime" | {
      startsWith: string;
  } | {
      endsWith: string;
  };
  export interface ZodInvalidStringIssue extends ZodIssueBase {
      code: typeof ZodIssueCode.invalid_string;
      validation: StringValidation;
  }
  export interface ZodTooSmallIssue extends ZodIssueBase {
      code: typeof ZodIssueCode.too_small;
      minimum: number;
      inclusive: boolean;
      exact?: boolean;
      type: "array" | "string" | "number" | "set" | "date";
  }
  export interface ZodTooBigIssue extends ZodIssueBase {
      code: typeof ZodIssueCode.too_big;
      maximum: number;
      inclusive: boolean;
      exact?: boolean;
      type: "array" | "string" | "number" | "set" | "date";
  }
  export interface ZodInvalidIntersectionTypesIssue extends ZodIssueBase {
      code: typeof ZodIssueCode.invalid_intersection_types;
  }
  export interface ZodNotMultipleOfIssue extends ZodIssueBase {
      code: typeof ZodIssueCode.not_multiple_of;
      multipleOf: number;
  }
  export interface ZodNotFiniteIssue extends ZodIssueBase {
      code: typeof ZodIssueCode.not_finite;
  }
  export interface ZodCustomIssue extends ZodIssueBase {
      code: typeof ZodIssueCode.custom;
      params?: {
          [k: string]: any;
      };
  }
  ```
</details>

**Comment by andrewvo89** [comment]
Thanks @Jackman3005, are you able to post a reproducible sample? Perhaps the schema and the data sample data that would fail the parse? And lastly, what you expect the final zod-error string to produce?

**Comment by andrewvo89** [comment]
Hey @Jackman3005, I gave this issue a bit more thought and have released an updated version (v1.4.0) where the error transform function now gets the original ZodIssue passed down. Here is an example of how it can be used. I hope this solves your problem:

```ts
import { ErrorMessageOptions, generateErrorMessage } from "zod-error";

import z from "zod";

const issues: z.ZodIssue[] = [
  {
    code: "invalid_arguments",
    argumentsError: {
      issues: [
        {
          code: "invalid_type",
          expected: "object",
          received: "array",
          path: [0],
          message: "Expected object, received array",
        },
      ],
      name: "ZodError",
    },
    path: [0, "renderConfig"],
    message: "Invalid function arguments",
  },
];

const options: ErrorMessageOptions = {
  transform: ({ issue, errorMessage }) => {
    if (issue.code === "invalid_arguments") {
      return issue.argumentsError.issues
        .map((innerIssue) => innerIssue.message)
        .join(", ");
    }
    return errorMessage;
  },
};

const errorMessage = generateErrorMessage(issues, options);
```

**Comment by Jackman3005** [comment]
I think this will definitely help. I'll give it a go tomorrow hopefully. Sorry for not getting back with an example earlier.
